homeassistant/components/mobile_app/helpers.py

```diff
@@ -166,12 +167,12 @@ def webhook_response(
     )
 
 
-def device_info(registration: dict) -> dict:
+def device_info(registration: dict) -> DeviceInfo:
     """Return the device info for this registration."""
     return {
         "identifiers": {(DOMAIN, registration[ATTR_DEVICE_ID])},
         "manufacturer": registration[ATTR_MANUFACTURER],
         "model": registration[ATTR_MODEL],
-        "device_name": registration[ATTR_DEVICE_NAME],
```

Comment:
Keep this as `device_name` please, otherwise you will break all integrating apps that are depending on it.

Reply:
How would that be breaking? It is invalid device info?

Reply:
Yeah, this is not an issue. The reason the name was set correctly is because in `async_setup_entry` the name is passed to the right field. It actually doesn't need to be passed in here either, we could just lmit it to `identifiers` here too. But this is fine.